PathPlanning/AStar/A_Star_searching_from_two_side.py

```diff
@@ -0,0 +1,363 @@
+    return border_ary
+
+
+def get_path(org_list, goal_list, coordinate):
+    # get path from start to end
+    path_org: list = []
+    path_goal: list = []
+    ind = find_node_index(coordinate, org_list)
+    node = org_list[ind]
+    while node != org_list[0]:
+        path_org.append(node.coordinate)
+        node = node.parent
+    path_org.append(org_list[0].coordinate)
+    ind = find_node_index(coordinate, goal_list)
+    node = goal_list[ind]
+    while node != goal_list[0]:
+        path_goal.append(node.coordinate)
+        node = node.parent
+    path_goal.append(goal_list[0].coordinate)
+    path_org.reverse()
+    path = path_org + path_goal
+    path = np.array(path)
+    return path
+
+
+def random_coordinate(bottom_vertex, top_vertex):
+    # generate random coordinates inside maze
+    coordinate = [np.random.randint(bottom_vertex[0] + 1, top_vertex[0]),
+                  np.random.randint(bottom_vertex[1] + 1, top_vertex[1])]
+    return coordinate
+
+
+def draw(close_origin, close_goal, start, end, bound):
+    # plot the map
+    if not close_goal.tolist():  # ensure the close_goal not empty
+        # in case of the obstacle number is really large (>4500), the
+        # origin is very likely blocked at the first search, and then
+        # the program is over and the searching from goal to origin
+        # will not start, which remain the closed_list for goal == []
+        # in order to plot the map, add the end coordinate to array
+        close_goal = np.array([end])
+    plt.cla()
+    plt.gcf().set_size_inches(11, 9, forward=True)
+    plt.axis('equal')
+    plt.plot(close_origin[:, 0], close_origin[:, 1], 'oy')
+    plt.plot(close_goal[:, 0], close_goal[:, 1], 'og')
+    plt.plot(bound[:, 0], bound[:, 1], 'sk')
+    plt.plot(end[0], end[1], '*b', label='Goal')
+    plt.plot(start[0], start[1], '^b', label='Origin')
+    plt.legend()
+    plt.pause(0.0001)
+
+
+def draw_control(org_closed, goal_closed, flag, start, end, bound, obstacle):
+    """
+    control the plot process
+    flag == 0 : draw the searching process and plot path
+    flag == 1 or 2 : start or end is blocked, draw the border line
+    """
+    stop_loop = 0  # stop sign for the searching
+    org_closed_ls = node_to_coordinate(org_closed)
+    org_array = np.array(org_closed_ls)
+    goal_closed_ls = node_to_coordinate(goal_closed)
+    goal_array = np.array(goal_closed_ls)
+    if show_animation:  # draw the searching process
+        draw(org_array, goal_array, start, end, bound)
+    if flag == 0:
+        node_intersect = check_node_coincide(org_closed, goal_closed)
+        if node_intersect:  # a path is find
+            path = get_path(org_closed, goal_closed, node_intersect[0])
+            stop_loop = 1
+            print('Path is find!')
+            if show_animation:  # draw the path
+                plt.plot(path[:, 0], path[:, 1], '-r')
+                plt.title('Robot Arrived', size=20, loc='center')
+                plt.show()
```

Comment:
Could you please add `plt.pause(0.01)` before `plt.show()`? In my environment, the final red line does not show  without it.

Reply:
Sure!